src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -499,11 +506,10 @@ async def get_repository_pex(
             internal_only=request.internal_only,
             requirements=Lockfile(
                 file_path=chosen_resolve.lockfile_path,
-                file_path_description_of_origin=(
-                    f"the resolve `{chosen_resolve.name}` (from `[python].resolves`)"
-                ),
+                file_path_description_of_origin=chosen_resolve.description_of_origin,
                 resolve_name=chosen_resolve.name,
-                req_strings=request.requirements.req_strings,
+                # NB: A blank req_strings means install the entire lockfile
+                req_strings=FrozenOrderedSet([]),
```

Comment:
This would be good to document on the `Lockfile` class rather-than/in-addition-to here. We haven't done a great job with docstrings on these internal APIs, but this in particular could be surprising.

Reply:
Well, it's only true if the Lockfile is a PEX lockfile AFAICT. :dizzy_face: